### ISSUE TITLE
Specify the return document type for AngularFirestoreDocument.get()

### ISSUE THREAD

**Issue by zer09** [angular/angularfire#2361]
The return type of the ```AngularFirestoreDocument.get()``` is not set, this will cause the editor cant perform autocomplete when calling DocumentSnapshot.data() because it will just treat its return type, as firestore.DocumentData instead the real type.

### Version info

**Angular:** 9.0.7

**Firebase:** 7.11.0

**AngularFire:** 5.4.2

**Other (e.g. Ionic/Cordova, Node, browser, operating system):**
Node: v12.6
npm: 6.14.2
Browser: Latest chrome and firefox nightly.

### How to reproduce these conditions

```typescript
export class Sample {
  constructor(private afStore AngularFirestore) {}
  
  test(path: string)  {
    this.afStore.doc<SampleType>(path).get().pipe(take(1)).subscribe(doc => {
      console.log(doc.data()); // The real type of doc.data() is not infered by the editor.
    })
  }
}
```

### Expected behavior

![personalinfo](https://user-images.githubusercontent.com/780780/77301271-604e0000-6d2a-11ea-883c-b8a895dba07f.png)

The return type should be 
```typescript
Observable<firebase.firestore.DocumentSnapshot<firebase.firestore.SamlpeType>>
```

This will allow to infer the type of the ```doc.data()```

### Actual behavior

![documentdata](https://user-images.githubusercontent.com/780780/77300905-cbe39d80-6d29-11ea-9f13-ad2939720c79.png)

The return type is 
```typescript
Observable<firebase.firestore.DocumentSnapshot<firebase.firestore.DocumentData>>
```
With this the autocomplete will not work

### Possible Solution

Based on ```index.d.ts``` on firestore, it is implemented like this.
```typescript
export class DocumentSnapshot<T = DocumentData> {
  data(options?: SnapshotOptions): T | undefined;
}
```

I think setting the return type of get on [document.ts](https://github.com/angular/angularfire/blob/master/src/firestore/document/document.ts#L101) to 
```typescript
get(options?: firestore.GetOptions): Observable<firestore.DocumentSnapshot<T>>
```
will fix the issue.

**Comment by KingDarBoja** [comment]
The return type should be indeed a generic as the other methods inside the `AngularFirestoreDocument` class definition.

**Comment by jamesdaniels** [comment]
Should be addressed in `6.1.0-rc.1`
